datalad/distribution/install.py

```diff
@@ -186,41 +186,34 @@ def _install_subds_inplace(ds, path, relativepath, source, runner):
 
 
 def _fixup_submodule_dotgit_setup(ds, relativepath):
-    """Implementation of our current of .git in a subdataset"""
+    """Implementation of our current of .git in a subdataset
+
+    Each subdataset/module has its own .git directory where a standalone
+    repository would have it. No gitdir files, no symlinks.
+    """
     # move .git to superrepo's .git/modules, remove .git, create
     # .git-file
     path = opj(ds.path, relativepath)
-    subds_git_dir = opj(path, ".git")
-    ds_git_dir = get_git_dir(ds.path)
-    moved_git_dir = opj(ds.path, ds_git_dir,
-                        "modules", relativepath)
-    # safety net
-    if islink(subds_git_dir) \
-            and realpath(subds_git_dir) == moved_git_dir:
-        # .git dir is already moved and linked
-        # remove link to enable .git replacement logic below
-        os.remove(subds_git_dir)
-    else:
-        # move .git
-        from os import rename, listdir, rmdir
-        assure_dir(moved_git_dir)
-        for dot_git_entry in listdir(subds_git_dir):
-            rename(opj(subds_git_dir, dot_git_entry),
-                   opj(moved_git_dir, dot_git_entry))
-        assert not listdir(subds_git_dir)
-        rmdir(subds_git_dir)
-
-    # TODO: symlink or whatever annex does, since annexes beneath
-    #       might break
-    #       - figure out, what annex does in direct mode
-    #         and/or on windows
-    #       - for now use .git file on windows and symlink otherwise
-    if not on_windows:
-        os.symlink(relpath(moved_git_dir, start=path),
-                   opj(path, ".git"))
-    else:
-        with open(opj(path, ".git"), "w") as f:
-            f.write("gitdir: {moved}\n".format(moved=relpath(moved_git_dir, start=path)))
+    subds_dotgit = opj(path, ".git")
+    src_dotgit = get_git_dir(path)
+
+    if src_dotgit == '.git':
+        # this is what we want
+        return
+
+    # what we have here is some kind of reference, remove and
+    # replace by the target
```

Comment:
sure... let's just do it then.. I think that my yesterday's demo has failed because of that gory git issue: http://git-annex.branchable.com/bugs/git-annex_confuses_Git_with_nested_submodules/ :
```shell
$> git annex get bold.nii.gz 
get bold.nii.gz (from datalad-archives...) 
git-annex: git: createProcess: runInteractiveProcess: chdir: does not exist (No such file or directory)
2016-05-26 14:01:51,345 [ERROR] stderr| git-annex: git: createProcess: runInteractiveProcess: chdir: does not exist (No such file or directory) (cmd.py:351)
2016-05-26 14:01:51,345 [ERROR] Failed to run ['git-annex', 'get', '--key', 'MD5E-s2527262329--bd3ea399057c529b37b09dcecec1ca60.0raw.tgz'] under '/mnt/btrfs/datasets/datalad/test-crawled2/datasets.datalad.org/openfmri/ds000001'. Exit code=1. out= err=git-annex: git: createProcess: runInteractiveProcess: chdir: does not exist (No such file or directory)
|  (cmd.py:295)
2016-05-26 14:01:51,345 [ERROR] Failed to fetch MD5E-s2527262329--bd3ea399057c529b37b09dcecec1ca60.0raw.tgz containing MD5E-s47241449--433b12536427334ded8e10eeb4a62d00.nii.gz: CommandError: command '['git-annex', 'get', '--key', 'MD5E-s2527262329--bd3ea399057c529b37b09dcecec1ca60.0raw.tgz']' failed with exitcode 1.
| Failed to run ['git-annex', 'get', '--key', 'MD5E-s2527262329--bd3ea399057c529b37b09dcecec1ca60.0raw.tgz'] under '/mnt/btrfs/datasets/datalad/test-crawled2/datasets.datalad.org/openfmri/ds000001'. Exit code=1. out= err=git-annex: git: createProcess: runInteractiveProcess: chdir: does not exist (No such file or directory)
|  (base.py:321)
                        
  Unable to access these remotes: datalad-archives

  Try making some of these repositories available:
  	899f0347-0888-48ef-91b6-bac213ca8cef -- [datalad-archives]
   	c8bd3d05-33d4-4b59-9d53-ca7efbdcdd13 -- yoh@smaug:/mnt/btrfs/datasets/datalad/crawl/openfmri/ds000001
failed
git-annex: get: 1 failed
(venv-tests)1 10166 ->1 [1].....................................:Thu 26 May 2016 02:01:51 PM EDT:.
(git)smaug:…atasets.datalad.org/openfmri/ds000001[master]sub001/BOLD/task001_run001
$> ls -l ../../../.git -d
lrwxrwxrwx 1 yoh datalad 24 May 24 15:25 ../../../.git -> ../.git/modules/ds000001/
```
looks like it?